Packet++/header/TcpReassembly.h

```diff
@@ -118,22 +123,16 @@ namespace pcpp
 		{}
 
 		/**
-		 * Set startTime of Connection
-		 * @param[in] startTimeValue integer value
+		 * Set the start time of the connection
+		 * @param[in] startTimeValue timestamp value
 		 */
-		void setStartTime(const timeval& startTimeValue)
-		{
-			startTime = startTimeValue;
-		}
+		void setStartTime(const std::chrono::time_point<std::chrono::system_clock>& startTimeValue);
 
 		/**
-		 * Set endTime of Connection
-		 * @param[in] endTimeValue integer value
+		 * Set the end time of the connection
+		 * @param[in] endTimeValue timestamp value
 		 */
-		void setEndTime(const timeval& endTimeValue)
-		{
-			endTime = endTimeValue;
-		}
+		void setEndTime(const std::chrono::time_point<std::chrono::system_clock>& endTimeValue);
```

Comment:
Why are the timepoints locked to `system_clock`? Generally I would prefer `steady_clock` for being monotonic, or `high_resolution_clock` for being the clock with the highest resolution on the system. Ideally, it should be templated, but that might be more trouble than its worth.

Reply:
What we need here is a data structure that holds a timestamp in nanosecond precision. The timestamp itself comes from the packet: https://github.com/seladb/PcapPlusPlus/blob/634a78654a94805298fde5850670d5b7788aaeda/Packet%2B%2B/src/TcpReassembly.cpp#L142

Does it matter if it's `system_clock`, `steady_clock` or `high_resolution_clock`? 🤔

Reply:
Not sure tbh, but there are a some methods of timepoint that depend on the passed clock type. It probably won't really affect the current usage, as nowhere is `Clock::now()` called. so idk.

Reply:
I tested it, apparently `system_clock` doesn't have nanosecond precision on all platforms, however, `high_resolution_clock` does so I think it's safe to use

Reply:
I mean we whould explicit the duration like this: 

using time_point_ns = std::chrono::time_point<std::chrono::high_resolution_clock, std::chrono::nanoseconds>;